niri-config/src/lib.rs

```diff
@@ -3937,6 +3957,139 @@ mod tests {
             .unwrap()
     }
 
+    #[test]
+    fn parse_per_axis_scroll_touchpad() {
+        let parsed = do_parse(
+            r#"
+            input {
+                touchpad {
+                    natural-scroll
+                    scroll-factor 1.5
+                    scroll-factor-horizontal 2.0
+                    scroll-factor-vertical -1.5
+                }
+            }
+            "#,
+        );
+
+        let touchpad = &parsed.input.touchpad;
+        assert_eq!(touchpad.natural_scroll, true);
+        assert_eq!(touchpad.scroll_factor, Some(FloatOrInt(1.5)));
+        assert_eq!(touchpad.scroll_factor_horizontal, Some(FloatOrInt(2.0)));
+        assert_eq!(touchpad.scroll_factor_vertical, Some(FloatOrInt(-1.5)));
```

Comment:
I prefer `assert_debug_snapshot!()` tests because they can update automatically when changing the config structure. Please convert these to just call `assert_debug_snapshot!()` like the main parsing test. Then call `cargo insta test --accept --all` to fill it in.

Reply:
I've added `assert_debug_snapshot!()` tests as suggested.